src/main/java/net/nikr/eve/jeveasset/gui/shared/table/EnumTableFormatAdaptor.java

```diff
@@ -441,17 +466,62 @@ public String getColumnToolTip(final int i) {
 
 	@Override public Object getColumnValue(final Q e, final int i) {
 		try {
-			return getColumn(i).getColumnValue(e);
+			return getColumnValue(e, getColumn(i));
 		} catch (IndexOutOfBoundsException ex) {
 			return null;
 		}
 	}
+
 	public Object getColumnValue(final Q e, final String columnName) {
-		EnumTableColumn<Q> column = orderColumnsName.get(columnName);
-		if (column != null)  {
+		return getColumnValue(e, orderColumnsName.get(columnName));
+	}
+
+	public Object getColumnValue(final Q e, final EnumTableColumn<Q> column) {
+		if (column == null) { //Better safe than sorry
+			return null;
+		}
+		Object object = column.getColumnValue(e);
+		if (object instanceof Formula) {
+			Formula formula = (Formula) object;
+			Object value = formula.getValues().get(e);
+			if (value == null) {
+				value = eval(formula, e);
+				formula.getValues().put(e, value);
+			}
+			return value;
+		} else {
 			return column.getColumnValue(e);
+		}
+	}
+
+	private Object eval(Formula formula, Q e) {
+		Expression expression = formula.getExpression();
+		for (T t : enumClass.getEnumConstants()) {
+			if (Number.class.isAssignableFrom(t.getType())) {
+				Number number = (Number) t.getColumnValue(e);
+				if (number == null) { //Handle null
+					expression.setVariable(JFormulaDialog.getHardName(t), new BigDecimal(0));
+					continue;
+				}
+				expression.setVariable(JFormulaDialog.getHardName(t), new BigDecimal(number.toString()));
+			} else if (NumberValue.class.isAssignableFrom(t.getType())) {
+				NumberValue numberValue = (NumberValue) t.getColumnValue(e);
+				if (numberValue == null) { //Handle null
+					expression.setVariable(JFormulaDialog.getHardName(t), new BigDecimal(0));
+					continue;
+				}
+				Number number = numberValue.getNumber();
+				if (number == null) { //Handle null
+					expression.setVariable(JFormulaDialog.getHardName(t), new BigDecimal(0));
+					continue;
+				}
+				expression.setVariable(JFormulaDialog.getHardName(t), new BigDecimal(number.toString()));
+			}
+		}
+		if (expression.isBoolean()) {
+			return expression.eval().doubleValue() > 0 ? "True" : "False";
```

Comment:
Floating point in Java always makes me a little wary with compactors. Should we use Double.compare() here instead?

Reply:
The expression can be both numeric and logical, isBoolean mean it's a logical expression.
eval() will return a BigDecimal, that is `0` if `false` and `1` if `true`.
We're return a String since it's used for the column value (If we return Boolean it will be rendered as a checkbox).
I wanted that I wanted the String value, so it's return "True" or "False".

Reply:
It was in reference to the "expression.eval().doubleValue() > 0" to "Double.compare(expression.eval().doubleValue(), 0.0) > 0". I know that you can have "rounding" issues with some floating point objects in Java but maybe that's more of an issue with ==

Reply:
I will take a look and see if I can improve this.